### PR TITLE
feat: allow runtime configuration

### PR DESCRIPTION
**Pull request by dcoa**
**Description**

This PR is part of the work to make it possible to configure the frontend applications at runtime (you can referer to this https://github.com/openedx/frontend-wg/issues/103).

**Changes**

 - Update frontend-platform to version 2.5.0
 - Use Helmet library to change the favicon in runtime.

Screenshot 
auth-logo

Values come from runtime configuration 
auth-response

**How to test**

- To allow runtime configuration set a new environment variables `MFE_CONFIG_API_URL` and `APP_ID` in the env file and add the api url ( To test this you can use the API from this https://github.com/openedx/edx-platform/pull/30473 or use an external tool to mock the API response).
- The API should respond with a JSON with the config values, something like:
```json
{
  "SITE_NAME": "Test Site",
  "LOGO_URL": "https://testimage.com/logo.svg",
  "LOGO_TRADEMARK_URL": "https://testimage.com/logo.svg",
  "LOGO_WHITE_URL": "https://testimage.com/logo.svg",
  "FAVICON_URL": "https://testimage.com/favicon.ico",
}
```
- The initialize process should work normally.
_Note: You can combine buildtime and runtime configuration_

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @dcoa!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by arbrandes** [comment]
@zainab-amir, mind reviewing this one?  It's required if we want to include it in Tutor/Olive.

**Comment by arbrandes** [comment]
@zainab-amir, thanks! 👍🏼

**Comment by zainab-amir** [comment]
@dcoa the changes look good. Kindly rebase the branch and I'll re-review it.

File: src/MainApp.jsx [hunk 1]
```diff
@@ -20,6 +22,9 @@ registerIcons();
 
 const MainApp = () => (
   <AppProvider store={configureStore()}>
+    <Helmet>
+      <link rel="shortcut icon" href={getConfig().FAVICON_URL} type="image/x-icon" />
+    </Helmet>
```
Comment (zainab-amir) [inline src/MainApp.jsx:27]: Should the favicon be removed from here then: https://github.com/openedx/frontend-app-authn/blob/e2899e66d107ef1625b7c6a971f032a71822d634/public/index.html#L7

**Comment by dcoa** [comment]
Hi @zainab-amir, I already rebase the branch and made the suggested change :)

**Review by zainab-amir — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@dcoa 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-authn&entry.867055334=https://github.com/openedx/frontend-app-authn/pull/654&entry.1484655318=https://github.com/dcoa&entry.752974735=2022-10-14+19:04&entry.1917517419=2022-11-30+06:13&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**